DOC: fix flake8 issue in groupby.rst

LJArendse opened the pull request:
Fixed flake8 issue in groupby.rst, see Issue #24178 for more info.

The following flake8 'errors' have been fixed:
- doc/source/groupby.rst:72:18: F821 undefined name 'obj'
- doc/source/groupby.rst:72:30: F821 undefined name 'key'
- doc/source/groupby.rst:73:18: F821 undefined name 'obj'
- doc/source/groupby.rst:73:30: F821 undefined name 'key'
- doc/source/groupby.rst:74:18: F821 undefined name 'obj'
- doc/source/groupby.rst:74:31: F821 undefined name 'key1'
- doc/source/groupby.rst:74:37: F821 undefined name 'key2'
- doc/source/groupby.rst:1306:42: F821 undefined name 'report_func'
- doc/source/groupby.rst:242:15: E225 missing whitespace around operator
- doc/source/groupby.rst:242:15: E999 SyntaxError: invalid syntax
- doc/source/groupby.rst:242:19: E225 missing whitespace around operator

LJArendse committed:
DOC: Remove doc/source/groupby.rst from flake8-rst exclude section in setup.cfg (#24178)

LJArendse committed:
DOC: Fix E225 and E999 caused by .<TAB> with # noqa: E225, E999

LJArendse committed:
DOC: Fix doc/source/groupby.rst:1306:42: F821 undefined name 'report_func' (#24178)

LJArendse committed:
DOC: Fix the following 'errors' (#24178):
doc/source/groupby.rst:72:18: F821 undefined name 'obj'
doc/source/groupby.rst:72:30: F821 undefined name 'key'
doc/source/groupby.rst:73:18: F821 undefined name 'obj'
doc/source/groupby.rst:73:30: F821 undefined name 'key'
doc/source/groupby.rst:74:18: F821 undefined name 'obj'
doc/source/groupby.rst:74:31: F821 undefined name 'key1'
doc/source/groupby.rst:74:37: F821 undefined name 'key2'

LJArendse committed:
Bring branch up to date Merge master

LJArendse committed:
DOC: Remove groupby from flake8-rst section exclusions (redo after merge with master)

jreback commented on doc/source/groupby.rst:
```diff
@@ -1301,11 +1315,15 @@ Piping can also be expressive when you want to deliver a grouped object to some
 arbitrary function, for example:
 
 .. code-block:: python
+   def mean(groupby):
```
need a blank line here

datapythonista replied:
also here, `.. code-block:: python` requires the `>>>` formatting, and the output, use the `ipython` directive (I know this was wrong before)

LJArendse replied:
@jreback Added blank line, see 28eb57f0f731b72e901158161792a0d364c1a661

jreback commented on doc/source/groupby.rst:
```diff
@@ -62,16 +62,30 @@ See the :ref:`cookbook<cookbook.grouping>` for some advanced strategies.
+* grouped = obj.groupby(key, axis='columns')
+* grouped = obj.groupby([key1, key2])
 
 .. code-block:: python
```
this could be an ipython:: python block

LJArendse replied:
This has been reverted, see b80ff177293437d0a306127e407ba8c671dd6b4f

datapythonista commented on doc/source/groupby.rst:
```diff
@@ -62,16 +62,30 @@ See the :ref:`cookbook<cookbook.grouping>` for some advanced strategies.
 -------------------------------
 
-pandas objects can be split on any of their axes. The abstract definition of
+Pandas objects can be split on any of their axes. The abstract definition of
```
Can you revert this please? Besides that `pandas` as a brand is always lower case, we try to avoid unrelated changes in PRs, so in case there are problems, it's easier to see where they are coming from, revert only the failing commit without unrelated reversals...

Surely you can propose those changes in a separate PR, to improve the changes.

LJArendse replied:
This has been reverted, see b80ff177293437d0a306127e407ba8c671dd6b4f

datapythonista commented on doc/source/groupby.rst:
```diff
@@ -62,16 +62,30 @@ See the :ref:`cookbook<cookbook.grouping>` for some advanced strategies.
+
+* grouped = obj.groupby(key)
+* grouped = obj.groupby(key, axis='columns')
+* grouped = obj.groupby([key1, key2])
```
Again, I'd prefer to not have the text changes in the PR to fix the flake8 issues. I know it's more complicated to do them in two PRs, but it helps to keep things under control in a big project like pandas. Also, I find the new sentence a bit difficult to read: "To create a GroupBy  object, see below (more on what the GroupBy object is later)."

LJArendse replied:
This has been reverted, see b80ff177293437d0a306127e407ba8c671dd6b4f

datapythonista commented on doc/source/groupby.rst:
```diff
@@ -62,16 +62,30 @@ See the :ref:`cookbook<cookbook.grouping>` for some advanced strategies.
+        columns=('class', 'order', 'max_speed')
+    )
+
+    grouped = df.groupby('class')
```
Besides this needs to be an `.. ipython:: python` block (otherwise the `>>>` and the output needs to be shown, please show the content of `df` after defining it, so it's easier to understand what the groupby is doing, and leave the comment saying that the default axis is index

LJArendse replied:
I have changed codeblock to```.. ipython:: python```, printed out the content of ```df```, and left the comment about default axis. (see c30f4567b60298836f867515f0f761da8807a3f0)

datapythonista commented on doc/source/groupby.rst:
```diff
@@ -1301,11 +1315,15 @@ Piping can also be expressive when you want to deliver a grouped object to some
 
 .. code-block:: python
+   def mean(groupby):
+       return(groupby.mean())
```
not sure why flake8 doesn't complain here, but in Python the correct way is `return groupby.mean()` without the brackets.

LJArendse replied:
@datapythonista I removed the brackets, see 00b03f0c5f2e1d05f1da98faafb37ab0575f3fcd

datapythonista reviewed (commented):
thanks for the fixes, added few comments

datapythonista commented:
@LJArendse do you have time to address comments of the reviews?

LJArendse commented:
@datapythonista Hi, yes I do. Sorry for the 'radio silence'.

LJArendse committed:
Revert "DOC: Remove groupby from flake8-rst section exclusions (redo after merge with master)"

This reverts commit dec7e639085bc2eb9a8800c9974d285471c59e58.

LJArendse committed:
Revert "DOC: Fix the following 'errors' (#24178):"

This reverts commit 41a2e4791e9a4835fb53ed1f754dc2ad1f9fcaea.

LJArendse committed:
Merge branch 'master' into doc-fix-flake8-issue-in-groupby.rst-#24178

LJArendse committed:
DOC: Add blank line

LJArendse committed:
DOC: Change code-block to ipython

LJArendse committed:
DOC: Remove return statement brackets

LJArendse committed:
DOC: Reapply code example fix from 41a2e4791e9a4835fb53ed1f754dc2ad1f9fcaea

LJArendse committed:
DOC: Remove doc/source/groupby.rst from setup.cfg

LJArendse committed:
Revert "DOC: Reapply code example fix from 41a2e4791e9a4835fb53ed1f754dc2ad1f9fcaea"

This reverts commit c30f4567b60298836f867515f0f761da8807a3f0.

LJArendse committed:
DOC: Fix the following flake8 issues
doc/source/groupby.rst:72:18: F821 undefined name 'obj'
doc/source/groupby.rst:72:30: F821 undefined name 'key'
doc/source/groupby.rst:73:18: F821 undefined name 'obj'
doc/source/groupby.rst:73:30: F821 undefined name 'key'
doc/source/groupby.rst:74:18: F821 undefined name 'obj'
doc/source/groupby.rst:74:31: F821 undefined name 'key1'
doc/source/groupby.rst:74:37: F821 undefined name 'key2'

datapythonista commented on doc/source/groupby.rst:
```diff
@@ -66,12 +66,23 @@ pandas objects can be split on any of their axes. The abstract definition of
+
+    df = pd.DataFrame(
+        [('bird', 'Falconiformes', 389.0),
+            ('bird', 'Psittaciformes', 24.0),
```
this indentation doesn't seem correct, the opening parenthesis should be at the same level

LJArendse committed:
DOC: Fix dataframe indentation

jreback commented:
thanks @LJArendse
